test/ResourceMatcher.test.js

```diff
@@ -598,6 +598,157 @@ describe("testResourceMatcher", () => {
         // translation of the plural adds or subtracts plural categories creating false matches
         expect(!actual).toBeTruthy();
     });
+});
+
+describe("resource-snake-case", () => {
+    test("defines the built-in declarative resource-matcher rule", () => {
+        const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
+
+        expect(rule.name).toBe("resource-snake-case");
+        expect(rule).toBeInstanceOf(ResourceMatcher);
+    });
+
+    test("sets the rule severity to 'error'", () => {
+        const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
+
+        expect(rule.severity).toBe("error");
+    });
+
+    test("sets the rule regexps to snake case variations", () => {
+        const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
 
+        expect(rule.re).toEqual(expect.arrayContaining([
+            /^\s*[a-z0-9]+(_[a-z0-9]+)+\s*$/dgu, // snake_case
+            /^\s*[A-Z0-9]+(_[A-Z0-9]+)+\s*$/dgu ,// SCREAMING_SNAKE_CASE
+            /^\s*[a-z0-9]+(_[A-Z0-9][a-z0-9]*)+\s*$/dgu, // camel_Snake_Case
+        ]));
+    });
+
+    test.each([
+        {name: "empty", source: ""},
+        {name: "undefined", source: undefined},
+        {name: "null", source: null},
+    ])("does not apply if source string is $name", ({source}) => {
+        const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
+        const resource = createTestResourceString({source, target: "does not matter"});
+
+        const result = rule.matchString({source: resource.source, target: resource.target, resource, file: resource.pathName});
+
+        expect(result).toBeUndefined();
+    });
+
+    // TODO @NatK: Fix in the next PR so that the rule does not run for falsy target values.
+    // it.each([
+    //     // {name: "empty", target: ""},
+    //     // {name: "undefined", target: undefined},
+    //     // {name: "undefined", target: null},
+    // ])("does not apply rule if target string is $name", ({target}) => {
+    //     const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
+    //     const resource = createTestResourceString({source: "snake_case", target});
+    //
+    //     const result = rule.matchString({source: resource.source, target: resource.target, resource, file: resource.pathName});
+    //
+    //     expect(result).toBeUndefined();
+    // });
+
+    test.each(
+        [
+            {name: "whitespace (solely)", source: " "},
+            {name: "text and whitespace", source: "snake case"},
+            {name: "snake case and text", source: "snake_case and text"},
+            {name: "screaming snake case and text", source: "SCREAMING_SNAKE_CASE and text"},
+            {name: "mixed case", source: "mixed_CASE"},
+        ]
```

Comment:
Also add a test for these sources for things that should not be considered to be snake case:

```
"_italic_",                           // simple markdown
"This is _italic_ text.",      // markdown
"word_",                            // trailing underscores do not make snake case
"word",                              // single word, no underscores
"345",                          // digits, no underscores
""                                        // empty string
"           "                             // string with only whitespace
"_"                                       // solo underscore
```